llvm/lib/Target/RISCV/GISel/RISCVLegalizerInfo.cpp

```diff
@@ -707,6 +707,14 @@ RISCVLegalizerInfo::RISCVLegalizerInfo(const RISCVSubtarget &ST)
 bool RISCVLegalizerInfo::legalizeIntrinsic(LegalizerHelper &Helper,
                                            MachineInstr &MI) const {
   Intrinsic::ID IntrinsicID = cast<GIntrinsic>(MI).getIntrinsicID();
+
+  const RISCVVIntrinsicsTable::RISCVVIntrinsicInfo *II =
+      RISCVVIntrinsicsTable::getRISCVVIntrinsicInfo(IntrinsicID);
+
+  if (II) {
+    return true;
+  }
```

Comment:
If `II` is otherwise unused, this should be:

```suggestion
  if (RISCVVIntrinsicsTable::getRISCVVIntrinsicInfo(IntrinsicID))
    return true;
```

Reply:
Addressed, I was thinking to use the intrinsic info to identify which rvv intrinsics need specific legalization, since the change is not in this pr, I changed this to the suggested form.